packages/application-shell/src/components/application-page-title/application-page-title.spec.tsx

```diff
@@ -0,0 +1,77 @@
+      expect(document.title).toBe(title);
+    });
+  });
+});
+
+describe.each`
+  route                                                                         | title
+  ${`/almond-40/dashboard`}                                                     | ${`Dashboard - ... chant Center`}
+  ${`/almond-40/products`}                                                      | ${`Products - a ... chant Center`}
+  ${`/almond-40/products/new`}                                                  | ${`Products - a ... chant Center`}
+  ${`/almond-40/products/<id>`}                                                 | ${`Products - a ... chant Center`}
+  ${`/almond-40/products/<id>/variants`}                                        | ${`Products - a ... chant Center`}
+  ${`/almond-40/products/<id>/variants/<id>`}                                   | ${`Products - a ... chant Center`}
+  ${`/almond-40/products/<id>/variants/<id>/prices`}                            | ${`Products - a ... chant Center`}
+  ${`/almond-40/orders/<id>`}                                                   | ${`Orders - alm ... chant Center`}
+  ${`/almond-40/settings/project/international`}                                | ${`Settings - a ... chant Center`}
+  ${`/almond-40/settings/project/stores`}                                       | ${`Settings - a ... chant Center`}
+  ${`/almond-40/settings/project/stores/<id>`}                                  | ${`Settings - a ... chant Center`}
+  ${`/almond-40/settings/product-types`}                                        | ${`Settings - a ... chant Center`}
+  ${`/almond-40/settings/product-types/<id>`}                                   | ${`Settings - a ... chant Center`}
+  ${`/almond-40/settings/product-types/<id>/attribute-definitions/detail/<id>`} | ${`Settings - a ... chant Center`}
+  ${`/account/projects`}                                                        | ${`Account - M ... chant Center`}
+  ${`/account/organizations`}                                                   | ${`Account - M ... chant Center`}
+  ${`/account/organizations/<id>/teams`}                                        | ${`Account - M ... chant Center`}
+  ${`/account/organizations/<id>/teams/<id>/permissions/<id>`}                  | ${`Account - M ... chant Center`}
+  ${`/account/organizations/<id>/custom-applications`}                          | ${`Account - M ... chant Center`}
+  ${`/account/organizations/<id>/custom-applications/owned/<id>`}               | ${`Account - M ... chant Center`}
+  ${`/account/organizations/<id>/custom-applications/installations/<id>`}       | ${`Account - M ... chant Center`}
+`(`'when the page title is too long: "$route"`, ({ route, title }) => {
+  it('should render a truncated readable page title', async () => {
+    render(
+      <Router history={memoryHistory(route)}>
+        <ApplicationPageTitle />
+      </Router>
+    );
+    await waitFor(() => {
+      expect(document.title).toBe(title);
+    });
+  });
```

Comment:
nit: perhaps we can extract this logic to a private helper function which can be used in all three tests ([here is an idea](https://gist.github.com/CarlosCortizasCT/ea4aab5614225e6becb19d45c1bb61b6))